File: samples/vehicle_gps_position_listener.py
import rclpy
from rclpy.node import Node
from rclpy.qos import QoSProfile, HistoryPolicy
from px4_msgs.msg import SensorGps

class VehicleGpsPositionListener(Node):

    def __init__(self):
        super().__init__('vehicle_global_position_listener')

        qos = QoSProfile(
            history=HistoryPolicy.RMW_QOS_POLICY_HISTORY_KEEP_LAST,
            depth=5,
            reliability=ReliabilityPolicy.RMW_QOS_POLICY_RELIABILITY_BEST_EFFORT,
            durability=DurabilityPolicy.RMW_QOS_POLICY_DURABILITY_VOLATILE
        )

        self.subscription = self.create_subscription(
            SensorGps,
            '/fmu/out/vehicle_gps_position',
            self.gps_callback,
            qos
        )

    def gps_callback(self, msg):
        print("\n" * 10)
        print("RECEIVED VEHICLE GPS POSITION DATA")
        print("==================================")
        print(f"ts: {msg.timestamp}")
        print(f"lat: {msg.lat}")
        print(f"lon: {msg.lon}")
        print(f"alt: {msg.alt}")
        print(f"alt_ellipsoid: {msg.alt_ellipsoid}")
        print(f"s_variance_m_s: {msg.s_variance_m_s}")
        print(f"c_variance_rad: {msg.c_variance_rad}")
        print(f"fix_type: {msg.fix_type}")
        print(f"eph: {msg.eph}")
        print(f"epv: {msg.epv}")
        print(f"hdop: {msg.hdop}")
        print(f"vdop: {msg.vdop}")
        print(f"noise_per_ms: {msg.noise_per_ms}")
        print(f"vel_m_s: {msg.vel_m_s}")
        print(f"vel_n_m_s: {msg.vel_n_m_s}")
        print(f"vel_e_m_s: {msg.vel_e_m_s}")
        print(f"vel_d_m_s: {msg.vel_d_m_s}")
        print(f"cog_rad: {msg.cog_rad}")
        print(f"vel_ned_valid: {msg.vel_ned_valid}")
        print(f"timestamp_time_relative: {msg.timestamp_time_relative}")
        print(f"time_utc_usec: {msg.time_utc_usec}")
        print(f"satellites_used: {msg.satellites_used}")
        print(f"heading: {msg.heading}")
        print(f"heading_offset: {msg.heading_offset}")

def main(args=None):
    print("Starting vehicle_global_position listener node...")
    rclpy.init(args=args)
    vehicle_gps_position_listener = VehicleGpsPositionListener()
    rclpy.spin(vehicle_gps_position_listener)
    vehicle_gps_position_listener.destroy_node()
    rclpy.shutdown()

if __name__ == '__main__':
    main()

File: samples/vehicle_gps_position_listener.cpp
#include <rclcpp/rclcpp.hpp>
#include <px4_msgs/msg/sensor_gps.hpp>

/**
 * @brief Vehicle GPS position uORB topic data callback
 */
class VehicleGpsPositionListener : public rclcpp::Node
{
public:
	explicit VehicleGpsPositionListener() : Node("vehicle_global_position_listener")
	{
		rmw_qos_profile_t qos_profile = rmw_qos_profile_sensor_data;
		auto qos = rclcpp::QoS(rclcpp::QoSInitialization(qos_profile.history, 5), qos_profile);
		
		subscription_ = this->create_subscription<px4_msgs::msg::SensorGps>("/fmu/out/vehicle_gps_position", qos,
		[this](const px4_msgs::msg::SensorGps::UniquePtr msg) {
			std::cout << "\n\n\n\n\n\n\n\n\n\n";
			std::cout << "RECEIVED VEHICLE GPS POSITION DATA"   << std::endl;
			std::cout << "=================================="   << std::endl;
			std::cout << "ts: "      << msg->timestamp    << std::endl;
			std::cout << "lat: " << msg->lat  << std::endl;
			std::cout << "lon: " << msg->lon << std::endl;
			std::cout << "alt: " << msg->alt  << std::endl;
			std::cout << "alt_ellipsoid: " << msg->alt_ellipsoid << std::endl;
			std::cout << "s_variance_m_s: " << msg->s_variance_m_s << std::endl;
			std::cout << "c_variance_rad: " << msg->c_variance_rad << std::endl;
			std::cout << "fix_type: " << msg->fix_type << std::endl;
			std::cout << "eph: " << msg->eph << std::endl;
			std::cout << "epv: " << msg->epv << std::endl;
			std::cout << "hdop: " << msg->hdop << std::endl;
			std::cout << "vdop: " << msg->vdop << std::endl;
			std::cout << "noise_per_ms: " << msg->noise_per_ms << std::endl;
			std::cout << "vel_m_s: " << msg->vel_m_s << std::endl;
			std::cout << "vel_n_m_s: " << msg->vel_n_m_s << std::endl;
			std::cout << "vel_e_m_s: " << msg->vel_e_m_s << std::endl;
			std::cout << "vel_d_m_s: " << msg->vel_d_m_s << std::endl;
			std::cout << "cog_rad: " << msg->cog_rad << std::endl;
			std::cout << "vel_ned_valid: " << msg->vel_ned_valid << std::endl;
			std::cout << "timestamp_time_relative: " << msg->timestamp_time_relative << std::endl;
			std::cout << "time_utc_usec: " << msg->time_utc_usec << std::endl;
			std::cout << "satellites_used: " << msg->satellites_used << std::endl;
			std::cout << "heading: " << msg->heading << std::endl;
			std::cout << "heading_offset: " << msg->heading_offset << std::endl;
		});
	}

private:
	rclcpp::Subscription<px4_msgs::msg::SensorGps>::SharedPtr subscription_;
};

int main(int argc, char *argv[])
{
	std::cout << "Starting vehicle_global_position listener node..." << std::endl;
	setvbuf(stdout, NULL, _IONBF, BUFSIZ);
	rclcpp::init(argc, argv);
	rclcpp::spin(std::make_shared<VehicleGpsPositionListener>());

	rclcpp::shutdown();
	return 0;
}
